### PR TITLE
Task/psgs 55 improve coverage

### PR DESCRIPTION
**Pull request by maciejbak85**
Another part of improving coverage, this time also added "checkIfContainsProperJsonKeys" for in memory api, which maybe does not improve coverage but will fail if there are some bugs ( like this with order my last PR before this )

### CODE REVIEW

File: src/test/scala/iog/psg/cardano/CardanoJpiSpec.scala [hunk 1]
```diff
@@ -51,7 +52,13 @@ class CardanoJpiSpec
...
   "POST /wallets" should "" in {
     api
-      .createRestore(wallet.name, "Pass9128!", mnemonicSentence.mnemonicSentence.toList.asJava, 5)
+      .createRestore(
+        wallet.name,
+        "Pass9128!",
+        mnemonicSentence.mnemonicSentence.toList.asJava,
+        mnemonicSecondFactor.mnemonicSentence.toList.asJava,
+        5
+      )
```
Comment (mcsherrylabs) [inline src/test/scala/iog/psg/cardano/CardanoJpiSpec.scala:61]: Like the style! :+1: Very readable.

File: src/test/scala/iog/psg/cardano/util/InMemoryCardanoApi.scala [hunk 2]
```diff
@@ -32,7 +32,11 @@ trait InMemoryCardanoApi {
   }
 
   implicit final class InMemoryExecutor[T](req: CardanoApiRequest[T]) {
-    def executeOrFail(): T = inMemoryExecutor.execute(req).futureValue.getOrElse(fail("Request failed."))
+    def executeOrFail(): T =
+      inMemoryExecutor.execute(req).futureValue match {
+        case Left(value)  => fail("Request failed: " + value.message)
+        case Right(value) => value
+      }
```
Comment (mcsherrylabs) [inline src/test/scala/iog/psg/cardano/util/InMemoryCardanoApi.scala:39]: :heart_eyes: style much more readable than on lhs.

Best to get out of the habit of using java string concatenation (+) and use s"${}"
It's a little more error prone and purists get worked up.

File: src/main/scala/iog/psg/cardano/TxMetadataOut.scala [hunk 3]
```diff
@@ -22,19 +22,19 @@ final case class TxMetadataOut(json: Json) {
 
       def extractStringField(cursor: ACursor): DecodingEither[MetadataValueStr] =
```
Comment (mcsherrylabs) [inline src/main/scala/iog/psg/cardano/TxMetadataOut.scala:23]: Lines 10 to 16, these should be defined in a companion object and not in every instance of the class.

**Review by mcsherrylabs — CHANGES_REQUESTED** [review]
Very Minor changes required.

File: src/main/scala/iog/psg/cardano/TxMetadataOut.scala (hunk 3 above)
Comment (mcsherrylabs) [reply src/main/scala/iog/psg/cardano/TxMetadataOut.scala:26]: Lines 10 to 16 Maciej, including the type definition.... it does not need to be declared on every instanciation.

Comment (maciejbak85) [reply src/main/scala/iog/psg/cardano/TxMetadataOut.scala:25]: missed type, I just pushed a commit I see that KeyVal type can be moved there too then, so moved it also

**Review by mcsherrylabs — APPROVED** [review]

**Pull request merged**